crates/topos-node/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+//! Temporary lib exposition for backward topos CLI compatibility
+use std::{path::PathBuf, process::ExitStatus};
+
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use opentelemetry::{global, sdk::metrics::controllers::BasicController};
+use process::Errors;
+use tokio::{
+    signal::{self, unix::SignalKind},
+    sync::mpsc,
+    task::JoinHandle,
+};
+use tokio_util::sync::CancellationToken;
+use topos_config::{
+    edge::command::BINARY_NAME,
+    genesis::Genesis,
+    node::{NodeConfig, NodeRole},
+};
+use topos_telemetry::tracing::setup_tracing;
+use topos_wallet::SecretManager;
+use tracing::{error, info};
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+mod process;
+
+#[allow(clippy::too_many_arguments)]
+pub async fn start(
+    verbose: u8,
+    no_color: bool,
+    otlp_agent: Option<String>,
+    otlp_service_name: Option<String>,
+    no_edge_process: bool,
+    node_path: PathBuf,
+    home: PathBuf,
+    config: NodeConfig,
+) -> Result<(), Box<dyn std::error::Error>> {
+    println!(
+        "⚙️ Reading the configuration from {}/config.toml",
+        node_path.display()
+    );
+
+    // Load genesis pointed by the local config
+    let genesis_file_path = home
+        .join("subnet")
+        .join(config.base.subnet.clone())
+        .join("genesis.json");
+
+    let genesis = match Genesis::new(genesis_file_path.clone()) {
+        Ok(genesis) => genesis,
+        Err(_) => {
+            println!(
+                "Could not load genesis.json file on path {} \n Please make sure to have a valid \
+                 genesis.json file for your subnet in the {}/subnet/{} folder.",
+                genesis_file_path.display(),
+                home.display(),
+                &config.base.subnet
+            );
+            std::process::exit(1);
```

Comment:
This is great. Should we also `log::error!` this same message?

Reply:
Maybe if the tracing setup is done, will create a little ticket for that